os/services/shell/shell-commands.c

```diff
@@ -734,7 +735,39 @@ shell_commands_init(void)
                                     echo_reply_handler);
 }
 /*---------------------------------------------------------------------------*/
-struct shell_command_t shell_commands[] = {
+void
+shell_command_set_register(struct shell_command_set_t *set)
+{
+  set->next = shell_command_sets_list;
+  shell_command_sets_list = set;
+}
+/*---------------------------------------------------------------------------*/
+void
+shell_command_set_deregister(struct shell_command_set_t *set)
+{
+  struct shell_command_set_t **prev;
+  struct shell_command_set_t *curr;
+
+  if(set == shell_command_sets_list) {
+    shell_command_sets_list = set->next;
+    set->next = NULL;
+    return;
+  }
+  prev = &shell_command_sets_list;
+  for(curr = shell_command_sets_list->next;
+      curr != NULL;
+      prev = &curr->next, curr = curr->next) {
+    if(curr == set) {
+      *prev = curr->next;
+      set->next = NULL;
+    }
+  }
+  if(curr != set) {
+    printf("Could not deregister command set: not found");
```

Comment:
Here I would just return an error code. Up to the user how to signal the error.

Reply:
and LOG_ERR perhaps

Reply:
Either log, or remove the printf. No strong preference

Reply:
IMO just return code is sufficient but feel free to add the log module